pkgs/tools/security/netexec/default.nix

```diff
@@ -0,0 +1,172 @@
+{ lib
+, fetchFromGitHub
+, python3
+, buildPythonPackage
+, fetchPypi
+}:
+let
+  python = python3.override {
+    packageOverrides = self: super: {
+      impacket = super.impacket.overridePythonAttrs {
+        version = "0.12.0.dev1";
+        src = fetchFromGitHub {
+          owner = "Pennyw0rth";
+          repo = "impacket";
+          rev = "d370e6359a410063b2c9c68f6572c3b5fb178a38";
+          hash = "sha256-Jozn4lKAnLQ2I53+bx0mFY++OH5P4KyqVmrS5XJUY3E=";
+        };
+      };
+      bloodhound-py = super.bloodhound-py.overridePythonAttrs (old: {
+        propagatedBuildInputs =
+          lib.lists.remove super.impacket old.propagatedBuildInputs
+          ++ [ self.impacket ];
```

Comment:
The point of creating `packageOverrides` is, that all packages in the consumed set use the overriden dependencies. So this construct is unnecessary.

Reply:
Hi @mweinelt! 
This `packageOverrides` was the way that I found to build `bloodhound-py` and `netexec` itself with different impacket versions. `netexec` uses a fork of impacket at a specific branch ([see here](https://github.com/Pennyw0rth/NetExec/blob/main/pyproject.toml#L46)) while `bloodhound-py` uses the original version, that is why I created and override but am not using it on `bloodhound-py`. Is there a better way to do it?

CC @thiagokokada